daemon/cmd/kube_proxy_replacement.go

```diff
@@ -107,6 +108,47 @@ func initKubeProxyReplacementOptions() (strict bool) {
 			log.Fatalf("Invalid value for --%s: %s", option.LoadBalancerDSRDispatch, option.Config.LoadBalancerDSRDispatch)
 		}
 
+		if option.Config.LoadBalancerRSSv4CIDR != "" {
+			ip, cidr, err := net.ParseCIDR(option.Config.LoadBalancerRSSv4CIDR)
+			if ip.To4() == nil {
+				err = fmt.Errorf("CIDR is not IPv4 based")
+			}
+			if err == nil {
+				if ones, _ := cidr.Mask.Size(); ones == 0 || ones >= 32 {
+					err = fmt.Errorf("CIDR length must be in (0,32)")
+				}
+			}
```

Comment:
No, the mask size `/x` would not be caught there.

Reply:
How do you mean /x? Does golang not detect `192.168.0.1/33` in `net.ParseCIDR()`?

Reply:
Now that it would, but `net.ParseCIDR()`would give you v4 or v6 back depending on your input, so it could just as well be a v6 addr with a mask fitting in that range, no?

Reply:
https://golang.org/src/net/ip.go?s=16533:16577#L725 it throws an error if the mask is out of the range of the ip length.

Reply:
OK, so the difference is specifically in handling `0.0.0.0/0` or `192.168.0.1/32`, both of which the current code would reject.

Reply:
Sorry, I misread your question, I thought dropping the To4() check (but then `2001:db8::/31` would pass for example). Yes, to drop the /0 and /32.

Reply:
> Now that it would, but net.ParseCIDR()would give you v4 or v6 back depending on your input, so it could just as well be a v6 addr with a mask fitting in that range, no?

IPv4 vs. IPv6 is already handled in the `To4()` case on line 113 yeah? https://github.com/cilium/cilium/pull/14276/files/d20a4290b375a8f5d0b5d5e7bbc8519a869a3f06#diff-c1e209552c5426ba2f2e01f043275558eb9f8ea65553210d2f9808de17b59d1eR113

Reply:
Why are /0 and /32 special?

Reply:
From out-of-band discussion: `/0` avoids undefined behaviour made by applying a `>> 32` shift. `/32` (or equivalent IPv6 mask) doesn't really make sense from a feature perspective. So fine to leave this as-is.